frontend/src/metabase/query_builder/components/CompareAggregations/components/OffsetInput/utils.ts

```diff
@@ -0,0 +1,62 @@
+import { t } from "ttag";
+
+import { pluralize } from "metabase/lib/formatting";
+import * as Lib from "metabase-lib";
+
+export const parsePeriodValue = (value: string): number | "" => {
+  const number = parseInt(value, 10);
+  return Number.isNaN(number) ? "" : Math.max(Math.abs(number), 1);
+};
+
+export const getLabel = (query: Lib.Query, stageIndex: number): string => {
+  const firstBreakout = Lib.breakouts(query, stageIndex)[0];
+
+  if (firstBreakout) {
+    const firstBreakoutColumn = Lib.breakoutColumn(
+      query,
+      stageIndex,
+      firstBreakout,
+    );
+
+    if (!Lib.isDate(firstBreakoutColumn)) {
+      return t`Row for comparison`;
+    }
+  }
+
+  return t`Previous period`;
+};
+
+export const getHelp = (query: Lib.Query, stageIndex: number): string => {
+  const firstBreakout = Lib.breakouts(query, stageIndex)[0];
+
+  if (!firstBreakout) {
+    return t`periods ago based on grouping`;
+  }
+
+  const firstBreakoutColumn = Lib.breakoutColumn(
+    query,
+    stageIndex,
+    firstBreakout,
+  );
+  const firstBreakoutColumnInfo = Lib.displayInfo(
+    query,
+    stageIndex,
+    firstBreakoutColumn,
+  );
+
+  if (!Lib.isDate(firstBreakoutColumn)) {
+    return t`rows above based on “${firstBreakoutColumnInfo.displayName}”`;
+  }
+
+  const bucket = Lib.temporalBucket(firstBreakout);
+
+  if (!bucket) {
+    return t`periods ago based on “${firstBreakoutColumnInfo.displayName}”`;
+  }
+
+  const bucketInfo = Lib.displayInfo(query, stageIndex, bucket);
+
+  return t`${pluralize(bucketInfo.displayName.toLowerCase())} ago based on “${
```

Comment:
Please use `describeTemporalInterval` instead. You can get the unit via `shortName`

Reply:
Those two give too much text but I was able to incorporate `describeTemporalUnit` here: 42397d0946a3aee911fa048c8009028b7c80a372